### ISSUE TITLE
Extra pages beside create, delete and edit

### ISSUE THREAD

**Issue by stochris** [marmelab/admin-on-rest#283]
Would it be possible to add support for extra pages for certain entities? 
In my case, it would be a reset password page, beside the normal "Edit user" page, so the admin can reset the password of any users.

**Comment by fzaninotto** [comment]
You can already do it via [a custom app](https://marmelab.com/admin-on-rest/CustomApp.html). But it's a good idea to provide the ability to add custom Routes in the `<Admin>` component.

**Comment by stochris** [comment]
I'm thinking an extra prop on the Resource component, that would describe any extra custom routes for that resource

**Comment by fzaninotto** [comment]
I don't understand, can you elaborate?

**Comment by stochris** [comment]
Well, maybe the `Resource` component could recieve another prop which describe the extra routes defined on the resource.
`
let extraRoutes = [
    {
        name: 'reset-password',
        component: myResetPasswordComponent
    }
]
<Resource name="users" list={myListComponent} extra={extraRoutes}>
</Resource>
`

**Comment by fzaninotto** [comment]
Why put it on the `Resource` component rather than in the `Admin` component? The latter would satisfy all needs IMO.

**Comment by stochris** [comment]
Ok, I can see your point. This would be in the form of another component beside 'Resource' ?

**Comment by fzaninotto** [comment]
No, it would be a prop of the `<Admin>` component:

```js
<Admin customRoutes={customRoutes}>
  <Resource name="users" list={UsersList} />
</Admin>
```

**Comment by fzaninotto** [comment]
Fixed by #299
